### PR TITLE
(2025) Torch dataset/dataloader

### PR DESCRIPTION
**Pull request by AlbertDominguez**
Here it is @ahillsley!

### Summary of changes
- Added torch dataset/dataloader for spiral data
- Shuffled around explanations to match new structure
- Refactored a few train/predict functions accordingly
- Changed XOR plot colors
- Minor style changes to the live training plots

### CODE REVIEW

File: exercise.ipynb [hunk 1]
```diff
@@ -643,11 +690,14 @@
     "batch_size = 10\n",
     "num_epochs = 1500\n",
     "\n",
+    "# Create the Dataloader associated to the training set. \n",
```
Comment (Cryaaa) [inline exercise.ipynb:693]: Maybe there should be one cell or some more text to explain what the difference between a torch dataset and a torch dataloader is? I know I struggled with this on my first go learning torch so I imagine others might profit from some guidance

Comment (AlbertDominguez) [reply exercise.ipynb:698]: Makes total sense! I've added that a bit further up in the last commit :)

**Review by ahillsley — COMMENTED** [review]
Looks good to me!
